clang/test/SemaSYCL/check-work-group-size-hint-host.cpp

```diff
@@ -0,0 +1,13 @@
+// RUN: %clang_cc1 -fsycl-is-host -fsyntax-only -verify %s
+
+// Host checks for the work_group_size_hint attribute.
+class Functor16x2x1 {
+public:
+  [[sycl::work_group_size_hint(16, 2, 1)]] void operator()() const {};
+};
+
+// The GNU spelling is deprecated in SYCL mode, but otherwise these attributes
+// have the same semantics.
+[[sycl::work_group_size_hint(4, 1, 1)]] void f4x1x1();
```

Comment:
Maybe it is a silly question, but what will happen if I type `[[sycl::work_group_size_hint...` in OpenCL C?

Reply:
In that specific case, would that even be parsed because of the namespace?

Reply:
Yeah, it seems it can't parse this spelling. I was wondering if we could just check if this is OpenCL compilation inside `handleWorkGroupSizeHint` instead of checking whether we have a namespace and what is the name of it, but after some thinking even though it seems we can, I find your solution more flexible in case there is more C++ like languages will use this attribute.

Reply:
A solution would be to check whether we're in SYCL so the check is "self contained". Otherwise if an OpenCL CXX appears this will break. (I just used the solution from `reqd_work_group_size`)